src/http/log.cr

```diff
@@ -0,0 +1,19 @@
+require "log"
+
+class HTTP::Client
+  Log = ::Log.for(self)
+
+  def_around_exec do |request|
+    emit_log({{request}})
+    yield
+  end
```

Comment:
```suggestion
  def_around_exec do |request|
    emit_log(request)
    yield request
  end
```

Reply:
I think that will only work if the block argument is named always `request`.

```
macro def_around_exec
  protected def around_exec(request)
                            ^-- this name might not be
...
def_around_exec do |request|
                    ^-- this one
end
```

Reply:
Ah, you are right. Then maybe we should't go this way because it's very unintuitive.